lib/src/widgets/primitives/pressable.dart

```diff
@@ -58,12 +60,108 @@ class PressableMixerWidget extends MixerWidget {
+      node = widget.focusNode ?? node;
+    }
+  }
+
+  FocusNode _createFocusNode() {
+    return FocusNode(debugLabel: '${widget.runtimeType}');
+  }
+
+  bool _hovering = false;
+  bool _pressing = false;
+  bool _shouldShowFocus = false;
+
+  bool get enabled => widget.onPressed != null || widget.onLongPressed != null;
+
   @override
   Widget build(BuildContext context) {
-    return InkWell(
-      child: BoxMixerWidget(
-        mixer,
-        child: child,
+    return MergeSemantics(
+      child: Semantics(
+        button: true,
+        enabled: enabled,
+        focusable: enabled && node.canRequestFocus,
+        focused: node.hasFocus,
+        child: FocusableActionDetector(
+          focusNode: node,
+          autofocus: widget.autofocus,
+          enabled: enabled,
+          onShowFocusHighlight: (v) {
+            if (mounted) setState(() => _shouldShowFocus = v);
+          },
+          onShowHoverHighlight: (v) {
+            if (mounted) setState(() => _hovering = v);
+          },
+          child: GestureDetector(
+            behavior: HitTestBehavior.opaque,
+            onTap: widget.onPressed,
+            onTapDown: (_) {
+              if (mounted) setState(() => _pressing = true);
+              // widget.onTapDown?.call();
+            },
+            onTapUp: (_) async {
+              // widget.onTapUp?.call();
+              if (!enabled) return;
+              await Future.delayed(const Duration(milliseconds: 100));
+              if (mounted) setState(() => _pressing = false);
+            },
+            onTapCancel: () {
+              // widget.onTapCancel?.call();
+              if (mounted) setState(() => _pressing = false);
+            },
+            onLongPressStart: (_) {
+              // widget.onLongPressStart?.call();
+              if (mounted) setState(() => _pressing = true);
+            },
+            onLongPressEnd: (_) {
+              // widget.onLongPressEnd?.call();
+              if (mounted) setState(() => _pressing = false);
+            },
+            child: () {
+              final disabled = widget.mixer.disabled;
+              final focused = widget.mixer.focused;
+              final hovering = widget.mixer.hovering;
+              final pressing = widget.mixer.pressing;
+
+              Mix? mix = () {
+                if (!enabled && disabled != null) return disabled.mix;
+                if (_pressing && pressing != null) return pressing.mix;
+                if (_hovering && hovering != null) return hovering.mix;
+                if (_shouldShowFocus && focused != null) return focused.mix;
+              }();
+
+              return BoxMixerWidget(
+                widget.mixer,
+                child: () {
+                  if (mix != null) {
+                    return BoxMixerWidget(
+                      Mixer.build(context, mix),
+                      child: widget.child,
+                    );
+                  }
+                  return widget.child;
```

Comment:
This, for some reason, is not being recognized. Means the box is not being painted if `pressed(Mix(...))` is used, for example.

Any clues?

Reply:
@bdlukaa Not sure exactly but looking at the code here are a few things. 
- I believe the MixWidget will have to be the one that is stateful.
- `Mixer.build()` allows to build the Mix based on the context, once the `mixer` is created I can be passed to a stateless widget and not modified.

Which means I think the logic need to be the following

```dart
/// In mix widget
Mix? gestureMix = () {
    if (!enabled && disabled != null) return disabled.mix;
    if (_pressing && pressing != null) return pressing.mix;
    if (_hovering && hovering != null) return hovering.mix;
    if (_shouldShowFocus && focused != null) return focused.mix;
}();

final mixer = Mixer.build(context, Mix.combine(mix, gestureMix));

```

Before running the `Mixer.build()` you will have to combine the gesture mix with the "original" mix passed into the pressable. The reason why is that this Mix, might contain styling information that you want to inherit, and use the gesture to `override`. 

Also, I believe there should be only 1 BoxMixer widget, and there is no need for the conditional check using the workflow I described above.

Reply:
Also one more thing, you are massing the `mix` into the attribute, we might want to consider passing an Attribute itself, similar to what is done on the `DarkModeAttribute`, you can find it here `dark_mode.dart`. 

This would allow passing an attribute, however, you can pass a mix by using the `apply` attribute, so the API could look something like this.

```dart
hovering(bgColor(Colors.green);
```
But would also allow for more robust implementations like this

```dart
hovering(apply(disabledMix, otherMix));
```

Reply:
I don't think that's a good idea. We can't use two `hovering` parameters because one will cancel the other

Reply:
Mixes can be composable, so they won't cancel out each other but will override properties. For example, a button can have an overall shape (rounded), margin and padding, together with a shadow, maybe you only want to override the shadow, or you want the color, but you don't want to compose all the basic shape again, this allows to easily override based on the order you call